### ISSUE TITLE
The automapper can drive its own timer and no caller has ever asked it to

### ISSUE THREAD

**Issue by malcyon** [malcyon/wish#390]
Does anything still need `AutomapBinding` to run its own polling timer, or is `drive=False` the only mode the program has?

## Why it matters

`AutomapBinding.__init__` takes `drive: bool = True`, and the `True` path builds and runs a `QTimer` that polls the machine itself. Every caller in the tree passes `False`:

- `wish/window.py:149` — the editor, which is the program a player runs
- `tools/mapmarker.py:257`
- every test that builds one

The real tick cadence comes from `wish/session.py`'s own `Session.timer` instead. So the default value of the parameter is the one nothing selects, and the code behind it has no exercised call site outside its own tests.

Found while building the bus-guard back-off for #375, when the question came up of where to put a hold-off and the obvious place turned out to be a mechanism nothing runs.

Nothing a player sees, which is why this is a question rather than a bug.

## What we know

The docstring around `automap/window.py:831` reads as though the two modes were deliberate: hosted (`drive=False`), trouble is raised rather than absorbed, against a standalone mode that absorbs it and keeps polling. That reads like an affordance for running the automapper on its own, without the editor around it — which may have been how it started, and may still be wanted.

**So this is not filed as dead code to delete.** A lever with a test and a docstring has been reported as dead code in this project before and was not. What is true is narrower: no caller selects it, and the parameter defaults to the unselected mode.

## What would settle it

One of three answers, and each leads somewhere different:

- **Nothing needs it** — the timer path and the parameter both go, and the default stops being a mode nobody runs.
- **Standalone use is still wanted** — then something should exercise it, because a path with no caller is a path that rots, and the default should probably still flip to match what every real caller passes.
- **It is a half-built feature** — then it wants its own issue saying what it was for.

Whoever answers this should also say whether `drive: bool = True` should default to `False` regardless, since a default that no call site in the tree selects is a trap for the next person who omits the argument.

Documentation: [`docs/99-one-window.md`](https://github.com/malcyon/wish/blob/main/docs/99-one-window.md)

**Comment by malcyon** [comment]
## Answer: (1) — nothing needs it, and the reason is on the record

`drive: bool = True` is a vestige of a real caller that was deliberately deleted, not code that never had one. Recommend removing the parameter, the timer-start branch, `_try_connect`, and the exception-absorbing branches in `tick()` and `shutdown()`; failing that, flip the default to `False` — but that is a design call for Donald, not mine to make.

### Call-site sweep, exhaustive

Every construction of `AutomapBinding` in the tree, found by grepping for `AutomapBinding(` and `AutomapBinding.__new__` (the debug-mode tests bypass `__init__` for a few unit tests):

- `wish/window.py:149` — the editor/map/combat host — `drive=False`
- `tools/mapmarker.py:257` — the standalone photography tool — `drive=False`, with its own comment: *"`drive=False` is how `wish/window.py` builds it: the host owns the connection and calls `tick()`, so nothing polls behind this script's back while `Session` is driving the game's menus."*
- `tests/test_strength.py:252`, `tests/test_combatlog.py` (5 sites), `tests/test_combat.py:266`, `tests/test_wronggame.py:328`, `tests/test_debugmode.py` (4 sites, one nested), `tests/test_automap.py:1711` — all `drive=False`

Every one, with no exception. No dynamic construction, no subclass, no `entry_points` in `pyproject.toml` beyond the single `wish = "wish.__main__:main"` script.

### What `drive=True` actually does differently

- `__init__` starts `self.timer` (`automap/window.py:768`) instead of leaving it built-but-stopped for a host to drive by calling `tick()` itself.
- `tick()` calls `self._try_connect()` when there is no target yet (`:877`), and on `NotConnected` or any other exception, absorbs it into `self._waiting`/`self.messages.say` instead of re-raising (`:899`, `:906`) — the "hosted (`drive=False`), trouble is raised rather than absorbed" line in `tick()`'s docstring at `:869` describes a real branch, not a stale comment.
- `shutdown()` calls `remember_geometry(self, self.settings)` and closes `self.mapper.target` (`:1362`, `:1366`) instead of leaving both to the host.

None of it is reached by any test: `_try_connect` has exactly two call sites in the whole tree, both inside `automap/window.py` itself (`:877`, its own `def`) — nothing calls it from a test or a driver. So the docstring's two-mode description is accurate about what the code does, but the "standalone" half of it has zero coverage anywhere.

### git history: it had a caller, and the caller was deleted on purpose

`git log -S"drive: bool"` finds one commit: `6a54460` ("Add the live automapper", 2026-08-20), the very first commit of this package. At that point `AutomapBinding` was `AutomapWindow`, a `QMainWindow` with its own `.ui` file, and `automap/__main__.py`'s `run()` built it with no `drive=` argument at all — taking the default `True`. `python -m automap` was that caller: it opened a real standalone window with its own timer, its own `QApplication`, attaching to VICE without any `wish` window around it.

Eight days later, commit `4049fdf` ("WIP: Unified UI integration and test fixes", 2026-08-28) turned `AutomapWindow` into the `QObject` `AutomapBinding` it is now — adding the `root` parameter, taking its widgets from a host's `.ui` instead of building its own — and deleted `automap/__main__.py` and `automap/ui_window.py` in the same commit. `run()` went with them. That is the one and only caller `drive=True` ever had, and it did not survive the merge into `wish/window.py`.

This was not an accident of that refactor — it was the plan. `docs/146-unified-ui.md`, written for that merge, says outright: *"Standalone entry points (`python -m editor`, `python -m automap`) are dropped"* and lists `automap/__main__.py` under `[DELETE or gut]`. `docs/99-one-window.md`, `docs/129-one-binary.md` and `pyproject.toml`'s own comment (*"`wish-cli`, `wish-editor` and `wish-automap` were dropped there"*) all say the same thing: one command, `wish`, and `wish --tab map` / `wish --svg` / `wish --forget` are what replaced the standalone map app. Confirmed live — `python -m automap --help` fails now with `No module named automap.__main__`.

So this is answer (1), reached through the (1) vs (3) distinction the issue itself draws: not a half-built feature waiting to be finished, but a fully-built one that was finished, shipped for eight days, and deliberately retired in favour of the hosted design — with the retirement documented by name. The default `True` and the four branches behind it are what got left behind when the caller was removed and nobody removed the code the caller alone exercised.

### Docstring drift found along the way, not part of this answer

Two docs still tell a reader to run the deleted command, and `tests/test_debugmode.py:832` still describes it as a live "entry point" — filed separately as #397 (docs/96-live-memory-automapper.md and docs/98-automap-notes.md still tell a reader to run `python -m automap`, which was deleted), so it does not get lost inside this comment.

### Checks

`ruff check automap/window.py` — clean. No code changed (this is a research answer, per the issue's own three-outcome list, outcomes (1) and (3) call for evidence and a proposal, not a patch), so no regression test to add and no fix to prove red-then-green. `pytest tests/test_automap.py`: 183 passed, 1 unrelated failure in `test_a_fight_disables_what_a_fight_forbids` (an `ActionBar` tooltip-text assertion) — caused by another agent's concurrent, uncommitted edit to `automap/actions.py` in this shared tree, not by anything here; confirmed by `git status` showing that file modified mid-session and the failing test never touching `AutomapBinding` or `drive`.

**Comment by malcyon** [comment]
Closing: answered above with the full call-site sweep and the git history of the one caller `drive=True` ever had. Removing the parameter and the four branches behind it (or flipping the default) is left as a decision for Donald, not built here.

**Comment by malcyon** [comment]
Removed. `AutomapBinding.__init__` no longer takes `drive`, and the four branches it gated are gone: the timer auto-start, `_try_connect()` (and the now-unused `connect` constructor parameter, `MonitorBusy`/`monitor_listening` imports it needed), the exception-absorbing half of `tick()`, and the `remember_geometry`/`target.close()` half of `shutdown()`.

**`shutdown()`'s two calls were deleted rather than made unconditional, deliberately, not by omission.** Both would have been wrong to keep:

- `remember_geometry(self, self.settings)` calls `self.saveGeometry()`, `.width()`, `.height()` on `self` -- and `AutomapBinding` is a `QObject`, not a `QWidget`, since the `4049fdf` refactor. It would have raised `AttributeError` the first time it ran, which is presumably why it never ran: `wish/window.py`'s own `closeEvent` already calls `remember_geometry(self, self.settings)` on the *`wish` window itself* before it calls `map.shutdown()`, and that is where geometry has been remembered all along.
- `self.mapper.target.close()` would have closed a connection `Session.close()` has already closed one line earlier in that same `closeEvent` -- `Session.detach()` is what actually owns `self.target`. The docstring's own words, "the connection is only ours to close when we opened it", were right; hosted, it never opened it.

So nothing here gains behaviour the host was not already providing. `tests/test_automap.py::test_shutdown_leaves_the_connection_and_the_geometry_to_the_host` pins both halves with a spy target and a settings-geometry snapshot, and `tests/test_preferences.py::test_the_window_remembers_its_size_and_opens_at_it` already covered the geometry half indirectly (it would have failed with `AttributeError` had the gate been removed the other way). `test_the_binding_no_longer_takes_a_drive_flag` pins the parameter itself being gone.

Every call site updated to stop passing `drive=False`: `wish/window.py`, `tools/mapmarker.py` (whose stale comment explaining the choice is also reworded), and the six test files that construct one directly.

`pytest tests/test_automap.py tests/test_combat.py tests/test_combatlog.py tests/test_debugmode.py tests/test_strength.py tests/test_wronggame.py tests/test_instance.py`: 503 passed, 1 failed -- `test_a_fight_disables_what_a_fight_forbids`, an `ActionBar` tooltip assertion from the concurrent #306 work landing in the same tree, untouched by anything here. 0 skipped. `ruff check .` and `tools/genui.py --check` both clean.

Docs corrected on #397 (docs/96-live-memory-automapper.md and docs/98-automap-notes.md still tell a reader to run `python -m automap`, which was deleted).
